src/build/arg/arg_value.rs

```diff
@@ -0,0 +1,77 @@
+use std::fmt::Display;
+
+/// TODO
+#[derive(Debug, Default, Clone)]
+pub struct ArgValue<'help> {
+    pub(crate) value: &'help str,
+    pub(crate) about: Option<&'help str>,
+    pub(crate) hidden: bool,
+}
+
+impl Display for ArgValue<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.value.contains(char::is_whitespace) {
+            write!(f, "{:?}", self.value)
+        } else {
+            write!(f, "{}", self.value)
+        }
+    }
+}
+
+impl<'help> From<&'help str> for ArgValue<'help> {
+    fn from(s: &'help str) -> Self {
+        Self::new(s)
+    }
+}
+
+impl<'help> From<&&'help str> for ArgValue<'help> {
+    fn from(s: &&'help str) -> Self {
+        Self::new(*s)
+    }
+}
```

Comment:
What case needs this?  Its not too common to derive for `&&str`

Reply:
I did that for this:
```
error[E0277]: the trait bound `ArgValue<'_>: From<&&str>` is not satisfied
  --> clap_generate/tests/value_hints.rs:14:18
   |
14 |                 .possible_values(&["bash", "fish", "zsh"]),
   |                  ^^^^^^^^^^^^^^^ the trait `From<&&str>` is not implemented for `ArgValue<'_>`
   |
   = help: the following implementations were found:
             <ArgValue<'help> as From<&'help str>>
   = note: required because of the requirements on the impl of `Into<ArgValue<'_>>` for `&&str`
   ```
   
   But probably there is a better way.

Reply:
I did that for this:
```
error[E0277]: the trait bound `ArgValue<'_>: From<&&str>` is not satisfied
  --> clap_generate/tests/value_hints.rs:14:18
   |
14 |                 .possible_values(&["bash", "fish", "zsh"]),
   |                  ^^^^^^^^^^^^^^^ the trait `From<&&str>` is not implemented for `ArgValue<'_>`
   |
   = help: the following implementations were found:
             <ArgValue<'help> as From<&'help str>>
   = note: required because of the requirements on the impl of `Into<ArgValue<'_>>` for `&&str`
   ```
   
   But probably there is a better way.

Reply:
Is the leading `&` needed, especially in 2021 edition?

Reply:
not here, but this would not be compatible with the derive, which requires a slice:
```rs
pub trait ArgEnum: Sized {
    /// All possible argument choices, in display order.
    const VARIANTS: &'static [&'static str];

    /// Parse an argument into `Self`.
    fn from_str(input: &str, case_insensitive: bool) -> Result<Self, String>;

    /// The canonical argument value.
    ///
    /// The value is `None` for skipped variants.
    fn as_arg(&self) -> Option<&'static str>;
}
```